# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/rocprofiler/no-aqlprofile.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- rocprofiler-rocm-6.3.3/src/util/hsa_rsrc_factory.cpp.vanilla	2025-03-21 18:53:16.988856313 +0200
+++ rocprofiler-rocm-6.3.3/src/util/hsa_rsrc_factory.cpp	2025-03-21 18:54:55.795188695 +0200
@@ -155,17 +155,6 @@
   if (kern_arg_pool_ == nullptr)
     CHECK_STATUS("Kern-arg memory pool is not found", HSA_STATUS_ERROR);
 
-  // Get AqlProfile API table
-  aqlprofile_api_ = {};
-#ifdef ROCP_LD_AQLPROFILE
-  status = LoadAqlProfileLib(&aqlprofile_api_);
-#else
-  status = hsa_api_.hsa_system_get_major_extension_table(HSA_EXTENSION_AMD_AQLPROFILE,
-                                                         hsa_ven_amd_aqlprofile_VERSION_MAJOR,
-                                                         sizeof(aqlprofile_api_), &aqlprofile_api_);
-#endif
-  CHECK_STATUS("aqlprofile API table load failed", status);
-
   // Get Loader API table
   loader_api_ = {};
   status = hsa_api_.hsa_system_get_major_extension_table(HSA_EXTENSION_AMD_LOADER, 1,
--- rocprofiler-rocm-6.3.3/src/tools/rocprofv2/CMakeLists.txt.vanilla	2025-03-21 18:56:02.611593991 +0200
+++ rocprofiler-rocm-6.3.3/src/tools/rocprofv2/CMakeLists.txt	2025-03-21 19:00:48.439497330 +0200
@@ -15,15 +15,6 @@
     TARGET hsa-runtime64::hsa-runtime64
     PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
 
-find_library(
-    AQLPROFILE_LIB "libhsa-amd-aqlprofile64.so"
-    HINTS ${CMAKE_PREFIX_PATH}
-    PATHS ${ROCM_PATH}
-    PATH_SUFFIXES lib)
-if(NOT AQLPROFILE_LIB)
-    message(FATAL_ERROR "AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
-endif()
-
 file(GLOB ROCPROFV2_SRC_FILES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
 
 add_executable(rocprofv2_bin ${ROCPROFV2_SRC_FILES})
--- rocprofiler-rocm-6.3.3/src/api/CMakeLists.txt.vanilla	2025-03-21 18:56:02.614748795 +0200
+++ rocprofiler-rocm-6.3.3/src/api/CMakeLists.txt	2025-03-21 19:01:02.956474016 +0200
@@ -51,15 +51,6 @@
     NO_DEFAULT_PATH REQUIRED)
 get_filename_component(HSA_RUNTIME_INC_PATH ${HSA_H} DIRECTORY)
 
-find_library(
-    AQLPROFILE_LIB "libhsa-amd-aqlprofile64.so"
-    HINTS ${CMAKE_PREFIX_PATH}
-    PATHS ${ROCM_PATH}
-    PATH_SUFFIXES lib)
-
-if(NOT AQLPROFILE_LIB)
-    message(FATAL_ERROR "AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
-endif()
 
 # ########################################################################################
 # Adding Old Library Files
--- rocprofiler-rocm-6.3.3/cmake_modules/rocprofiler_env.cmake.vanilla	2025-03-21 19:02:03.427297297 +0200
+++ rocprofiler-rocm-6.3.3/cmake_modules/rocprofiler_env.cmake	2025-03-21 19:02:47.855024026 +0200
@@ -80,9 +80,3 @@
     message(FATAL_ERROR "ROCM_ROOT_DIR is not found.")
 endif()
 
-find_library(
-    HSA_AMD_AQLPROFILE_LIBRARY
-    NAMES hsa-amd-aqlprofile64
-    HINTS ${CMAKE_PREFIX_PATH}
-    PATHS ${ROCM_ROOT_DIR}
-    PATH_SUFFIXES lib REQUIRED)
